src/web/app/pages-instructor/instructor-home-page/instructor-home-page.component.spec.ts

```diff
@@ -72,13 +123,294 @@ describe('InstructorHomePageComponent', () => {
   beforeEach(() => {
     fixture = TestBed.createComponent(InstructorHomePageComponent);
     component = fixture.componentInstance;
+    courseService = TestBed.inject(CourseService);
+    simpleModalService = TestBed.inject(SimpleModalService);
+    feedbackSessionsService = TestBed.inject(FeedbackSessionsService);
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
 
+  it('should get course candidates', () => {
+    component.courseTabModels = activeCourseTabModels;
+    fixture.detectChanges();
+
+    const courses: Course[] = component.courseCandidates;
+
+    expect(courses.length).toEqual(2);
+    expect(courses[0].courseId).toEqual('CS1231');
+    expect(courses[0].courseName).toEqual('Discrete Structures');
+  });
+
+  it('should expand the course tab model upon clicking', () => {
+    component.courseTabModels = activeCourseTabModels;
+    component.hasCoursesLoaded = true;
+    fixture.detectChanges();
+
+    const button: any = fixture.debugElement.nativeElement.querySelector('#course-tab-header');
+    button.click();
+    expect(component.courseTabModels[0].isTabExpanded).toBeTruthy();
+
+    button.click();
+    expect(component.courseTabModels[0].isTabExpanded).toBeFalsy();
+  });
+
+  it('should archive the entire course from the instructor', () => {
+    const archiveCourseTabModels: CourseTabModel[] = [
+      {
+        course: {
```

Comment:
I'm wondering whether it might be worthwhile extracting the course data here into a constant variable

Reply:
I agree. Will do a round of thorough refactoring for the addition.